docs/using-ngrok-with/k8s.md

```diff
@@ -0,0 +1,246 @@
+
+    ![ingress created](/img/howto/ingress-controller/k8s-ingress-app-1.png)
+
+1. Access your ingress URL (i.e. `my-awesome-k8s-cluster.ngrok.app`) to confirm the 2048 app is accessible from the internet:
+    
+    ![application public](/img/howto/ingress-controller/k8s-ingress-app-2.png)
+
+## Step 3: Add edge security to your app
+
+The ngrok Ingress Controler for Kubernetes provides custom resource definitions (CRDs) for additional edge features available in ngrok. In this example, we're expanding the Ingress Controller with Google OAuth to allow access only from users with the email domains `@acme.com` or `@ngrok.com` and to apply a circuit breaker to your app at 80%. These features are enforced at the ngrok edge, ensuring only authorized users can access your app.
+
+:::tip Notes:
+This example is very similar to the previous version, with the following changes:
+- Lines 40-42: Associates your Ingress Controller with the configuration `oauth-and-circuit-breaking` via annotation
+- Lines 56-72: Sets the edge configuration as a custom CRD (NgrokModuleSet).
+- Lines 62-67: Sets the circuit breaker module over 50% threshold.
+- Lines 68-72: Sets the OAuth module to allow access only for users with the email domains `@acme.com` or `@ngrok.com`.
+:::
+
+```yaml showLineNumbers
+apiVersion: v1
+kind: Service
+metadata:
+  name: game-2048
+spec:
+  ports:
+    - name: http
+      port: 80
+      targetPort: 80
+  selector:
+    app: game-2048
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: game-2048
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: game-2048
+  template:
+    metadata:
+      labels:
+        app: game-2048
+    spec:
+      containers:
+        - name: backend
+          image: alexwhen/docker-2048
+          ports:
+            - name: http
+              containerPort: 80
+---
+# Instead of running the k8s ingress on port 80
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+ name: game-2048-ingress
+ annotations:
+  # Associate edge with module configuration
+  k8s.ngrok.com/modules: oauth-and-circuit-breaking
+spec:
+ ingressClassName: ngrok
+ rules:
+   - host: my-awesome-k8s-cluster.ngrok.app
+     http:
+       paths:
+         - path: /
+           pathType: Prefix
+           backend:
+             service:
+               name: game-2048
+               port:
+                 number: 80
+---
+# Module configurations for Circuit Breaking and OAuth
+kind: NgrokModuleSet
+apiVersion: ingress.k8s.ngrok.com/v1alpha1
+metadata:
+  name: oauth-and-circuit-breaking
+modules:
+  circuitBreaker:
+    trippedDuration: 10s
+    rollingWindow: 10s
+    numBuckets: 10
+    volumeThreshold: 20
+    errorThresholdPercentage: "0.50"
+  oauth:
+    google:
+      emailDomains:
+      - acme.com
+      - ngrok.com
+---
+```
+
+1. Save your file and reapply the manifest file:
+    
+    ```bash
+    kubectl delete -f ngrok-manifest.yaml
+    kubectl apply -f ngrok-manifest.yaml
```

Comment:
No need to do a delete here, just an apply should skip the resources with no changes and update the ones that did saying they are configured

Reply:
nice!